pkg/sql/pgwire/testdata/pgtest/pgjdbc

```diff
@@ -72,3 +72,73 @@ ReadyForQuery
 {"Type":"CommandComplete","CommandTag":"BEGIN"}
 {"Type":"CommandComplete","CommandTag":"SAVEPOINT"}
 {"Type":"ReadyForQuery","TxStatus":"T"}
+
+send
+Query {"String": "COMMIT"}
+----
+
+until
+ReadyForQuery
+----
+{"Type":"CommandComplete","CommandTag":"COMMIT"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+
+send
+Query {"String": "DROP TABLE IF EXISTS t"}
+----
+
+until
+ReadyForQuery
+----
+{"Type":"CommandComplete","CommandTag":"DROP TABLE"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+
+send
+Query {"String": "CREATE TABLE IF NOT EXISTS t(a INT8 UNIQUE)"}
+----
+
+until
+ReadyForQuery
+----
+{"Type":"CommandComplete","CommandTag":"CREATE TABLE"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+
+
+# Test that a simple query in the middle of the extended protocol
+# causes the earlier statement to commit.
+send
+Parse {"Name": "S_4", "Query": "INSERT INTO t VALUES(1)"}
+Bind {"PreparedStatement": "S_4"}
+Execute
+Query {"String": "SELECT 1"}
```

Comment:
can you add a comment that the prepared statement is re-executed and we assume a constraint error from that? that confused me for a second.

Reply:
yeah for sure